### ISSUE TITLE
Error executing action `install` on resource 'apt_package[nodejs]'

### ISSUE THREAD

**Issue by kyron500** [rapid7/metasploitable3#515]
## Issue Description
I'm trying to build the Ubuntu14.04 Metasploitable3 VM using VirtualBox. The VM starts normally, without issues. The problem is, when in some point during the execution of the installation script I get the following message: **_virtualbox-iso:     Error executing action `install` on resource 'apt_package[nodejs]'_**

With error:
```
Error building the Vagrant box using Packer. Please check the output above for any error messages.
En C:\Users\Kyron\Documents\_metasploitable3-master\metasploitable3-master\build.ps1: 157 Carácter: 10
+          throw "Error building the Vagrant box using Packer. Please c ...
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Error building ...error messages.:String) [], RuntimeException
    + FullyQualifiedErrorId : Error building the Vagrant box using Packer. Please check the output above for any error
     messages.
```

## Host System
* OS: Windows10 20H2
* Packer Version: 1.5.5
* Vagrant Version: 2.2.14
* VirtualBox Version: 6.1.18

## Command Output
The trace after the error can be found here:
https://gist.github.com/kyron500/510ad5fe008623d4073879c3e99b8bf2

**Comment by robertstrom** [comment]
I am seeing the same when trying to build the ubuntu1404. I was able to successfully build the Win2k8 with this same system.

    virtualbox-iso:     ================================================================================
    virtualbox-iso:     Error executing action `install` on resource 'apt_package[nodejs]'
    virtualbox-iso:     ================================================================================
    virtualbox-iso:
    virtualbox-iso:     Mixlib::ShellOut::ShellCommandFailed
    virtualbox-iso:     ------------------------------------
    virtualbox-iso:     Expected process to exit with [0], but received '100'
    virtualbox-iso:     ---- Begin output of ["apt-get", "-q", "-y", "install", "nodejs=4.9.1-1nodesource1"] ----
    virtualbox-iso:     STDOUT: Reading package lists...
    virtualbox-iso:     Building dependency tree...
    virtualbox-iso:     Reading state information...
    virtualbox-iso:     The following NEW packages will be installed:
    virtualbox-iso:       nodejs
    virtualbox-iso:     0 upgraded, 1 newly installed, 0 to remove and 175 not upgraded.
    virtualbox-iso:     Need to get 8388 kB of archives.
    virtualbox-iso:     After this operation, 36.6 MB of additional disk space will be used.
    virtualbox-iso:     WARNING: The following packages cannot be authenticated!
    virtualbox-iso:       nodejs
    virtualbox-iso:     STDERR: E: There are problems and -y was used without --force-yes
    virtualbox-iso:     ---- End output of ["apt-get", "-q", "-y", "install", "nodejs=4.9.1-1nodesource1"] ----
    virtualbox-iso:     Ran ["apt-get", "-q", "-y", "install", "nodejs=4.9.1-1nodesource1"] returned 100
    virtualbox-iso:
    virtualbox-iso:     Resource Declaration:
    virtualbox-iso:     ---------------------
    virtualbox-iso:     # In /tmp/packer-chef-solo/local-mode-cache/cache/cookbooks/metasploitable/recipes/nodejs.rb
    virtualbox-iso:
    virtualbox-iso:      13: package 'nodejs'
    virtualbox-iso:
    virtualbox-iso:     Compiled Resource:
    virtualbox-iso:     ------------------
    virtualbox-iso:     # Declared in /tmp/packer-chef-solo/local-mode-cache/cache/cookbooks/metasploitable/recipes/nodejs.rb:13:in `from_file'
    virtualbox-iso:
    virtualbox-iso:     apt_package("nodejs") do
    virtualbox-iso:       package_name "nodejs"
    virtualbox-iso:       action [:install]
    virtualbox-iso:       default_guard_interpreter :default
    virtualbox-iso:       declared_type :package
    virtualbox-iso:       cookbook_name "metasploitable"
    virtualbox-iso:       recipe_name "nodejs"
    virtualbox-iso:     end
    virtualbox-iso:
    virtualbox-iso:     System Info:
    virtualbox-iso:     ------------
    virtualbox-iso:     chef_version=13.8.5
    virtualbox-iso:     platform=ubuntu
    virtualbox-iso:     platform_version=14.04
    virtualbox-iso:     ruby=ruby 2.4.3p205 (2017-12-14 revision 61247) [x86_64-linux]
    virtualbox-iso:     program_name=chef-solo worker: ppid=1094;start=06:16:10;
    virtualbox-iso:     executable=/opt/chef/bin/chef-solo
    virtualbox-iso:
    virtualbox-iso: Recipe: iptables::default
    virtualbox-iso:   * execute[rebuild-iptables] action run
    virtualbox-iso:     - execute /usr/sbin/rebuild-iptables
    virtualbox-iso:
    virtualbox-iso: Running handlers:
    virtualbox-iso: [2021-03-28T06:54:38+00:00] ERROR: Running exception handlers
    virtualbox-iso: [2021-03-28T06:54:38+00:00] ERROR: Running exception handlers
    virtualbox-iso: Running handlers complete
    virtualbox-iso: [2021-03-28T06:54:38+00:00] ERROR: Exception handlers complete
    virtualbox-iso: [2021-03-28T06:54:38+00:00] ERROR: Exception handlers complete
    virtualbox-iso: Chef Client failed. 175 resources updated in 38 minutes 28 seconds
    virtualbox-iso: [2021-03-28T06:54:38+00:00] FATAL: Stacktrace dumped to /tmp/packer-chef-solo/local-mode-cache/cache/chef-stacktrace.out
    virtualbox-iso: [2021-03-28T06:54:38+00:00] FATAL: Stacktrace dumped to /tmp/packer-chef-solo/local-mode-cache/cache/chef-stacktrace.out
    virtualbox-iso: [2021-03-28T06:54:38+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
    virtualbox-iso: [2021-03-28T06:54:38+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
    virtualbox-iso: [2021-03-28T06:54:38+00:00] ERROR: apt_package[nodejs] (metasploitable::nodejs line 13) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '100'
    virtualbox-iso: ---- Begin output of ["apt-get", "-q", "-y", "install", "nodejs=4.9.1-1nodesource1"] ----
    virtualbox-iso: STDOUT: Reading package lists...
    virtualbox-iso: Building dependency tree...
    virtualbox-iso: Reading state information...
    virtualbox-iso: The following NEW packages will be installed:
    virtualbox-iso:   nodejs
    virtualbox-iso: 0 upgraded, 1 newly installed, 0 to remove and 175 not upgraded.
    virtualbox-iso: Need to get 8388 kB of archives.
    virtualbox-iso: After this operation, 36.6 MB of additional disk space will be used.
    virtualbox-iso: WARNING: The following packages cannot be authenticated!
    virtualbox-iso:   nodejs
    virtualbox-iso: STDERR: E: There are problems and -y was used without --force-yes
    virtualbox-iso: ---- End output of ["apt-get", "-q", "-y", "install", "nodejs=4.9.1-1nodesource1"] ----
    virtualbox-iso: Ran ["apt-get", "-q", "-y", "install", "nodejs=4.9.1-1nodesource1"] returned 100
    virtualbox-iso: [2021-03-28T06:54:38+00:00] ERROR: apt_package[nodejs] (metasploitable::nodejs line 13) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '100'
    virtualbox-iso: ---- Begin output of ["apt-get", "-q", "-y", "install", "nodejs=4.9.1-1nodesource1"] ----
    virtualbox-iso: STDOUT: Reading package lists...
    virtualbox-iso: Building dependency tree...
    virtualbox-iso: Reading state information...
    virtualbox-iso: The following NEW packages will be installed:
    virtualbox-iso:   nodejs
    virtualbox-iso: 0 upgraded, 1 newly installed, 0 to remove and 175 not upgraded.
    virtualbox-iso: Need to get 8388 kB of archives.
    virtualbox-iso: After this operation, 36.6 MB of additional disk space will be used.
    virtualbox-iso: WARNING: The following packages cannot be authenticated!
    virtualbox-iso:   nodejs
    virtualbox-iso: STDERR: E: There are problems and -y was used without --force-yes
    virtualbox-iso: ---- End output of ["apt-get", "-q", "-y", "install", "nodejs=4.9.1-1nodesource1"] ----
    virtualbox-iso: Ran ["apt-get", "-q", "-y", "install", "nodejs=4.9.1-1nodesource1"] returned 100
    virtualbox-iso: [2021-03-28T06:54:39+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
    virtualbox-iso: [2021-03-28T06:54:39+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
==> virtualbox-iso: Provisioning step had errors: Running the cleanup provisioner, if present...
==> virtualbox-iso: Cleaning up floppy disk...
==> virtualbox-iso: Deregistering and deleting VM...
==> virtualbox-iso: Deleting output directory...
Build 'virtualbox-iso' errored after 1 hour 6 minutes: Error executing Chef: Non-zero exit status: 1

==> Wait completed after 1 hour 6 minutes

==> Some builds didn't complete successfully and had errors:
--> virtualbox-iso: Error executing Chef: Non-zero exit status: 1

**Comment by russdan** [comment]
To fix update`metasploitable3-master\chef\cookbooks\metasploitable\recipes\nodejs.rb`  from

`package 'nodejs'`

to 

```
package 'nodejs' do
  options '--force-yes'
end
```

this then passes `--force-yes` to the apt install and seems to resolve the issue

```
    virtualbox-iso: Recipe: metasploitable::nodejs
    virtualbox-iso:   * execute[add nodejs 4 repository] action run
    virtualbox-iso:     - execute curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -
    virtualbox-iso:   * apt_package[nodejs] action install
    virtualbox-iso:     - install version 4.9.1-1nodesource1 of package nodejs
```

**Comment by deargle** [comment]
Hrm, I wonder why it failed to authenticate the nodejs package, and what
other problems will come up later because of that.

**Comment by jmartin-r7** [comment]
Closed by #527
